packages/components/src/button/index.native.js

```diff
@@ -172,28 +182,33 @@ export function Button( props ) {
 			accessibilityRole={ 'button' }
 			accessibilityHint={ hint }
 			onPress={ onClick }
-			onLongPress={ onLongPress }
 			style={ containerStyle }
 			disabled={ isDisabled }
 			testID={ testID }
 		>
-			<View style={ buttonViewStyle }>
-				<View style={ { flexDirection: 'row' } }>
-					{ newIcon }
-					{ newChildren }
-					{ subscript && (
-						<Text
-							style={
-								isPressed
-									? styles.subscriptActive
-									: subscriptInactive
-							}
-						>
-							{ subscript }
-						</Text>
-					) }
+			<LongPressGestureHandler
+				minDurationMs={ 500 }
+				maxDist={ 150 }
+				onHandlerStateChange={ longPressHandler }
+			>
```

Comment:
I decided to use the component `LongPressGestureHandler` instead of the Long-press gesture due to performance issues. I haven't checked further this, but using the gesture, although [it's recommended by the library](https://docs.swmansion.com/react-native-gesture-handler/docs/gesture-handlers/api/longpress-gh), impacts heavily on performance.